### PR TITLE
Justified revision

### PR DESCRIPTION
**Pull request by paologalligit**
# Description

This PR refers to this [ticket](https://github.com/vechain/protocol-board-repo/issues/260). It introduces a new revision named **justified**. The meaning of this revision is the same Ethereum is using (see [here](https://ethos.dev/beacon-chain), in geth the justified revision is referred to as _safe_). The chain is composed of epochs (each epoch is composed of 180 blocks) whose starting point is called _checkpoint_. When an epoch ends, if its checkpoint has garnered a ⅔ supermajority, the checkpoint gets justified; this happens every 30 minutes.

Example:

-- current best block
 |
 |
-- checkpoint
 |
 |
 |
 |
 |
-- justified checkpoint
 |
 |
 |
 |
 |
-- finalized checkpoint

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- running thor unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Comment by leszek-vechain** [comment]
can we add test for justified in e2e ? (we can open PR for now and just merge it when needed, before we forget about it)

File: api/accounts/accounts.go [hunk 1]
```diff
@@ -31,15 +31,15 @@ type Accounts struct {
...
-	bft          bft.Finalizer
+	bft          bft.CommitLevel
 }
 
 func New(
 	repo *chain.Repository,
 	stater *state.Stater,
 	callGasLimit uint64,
 	forkConfig thor.ForkConfig,
-	bft bft.Finalizer,
```
Comment (libotony) [inline api/accounts/accounts.go:42]: For the name convention, I would consider verb + `er` suffix is better, wdyt? Can you come up better options?

Comment (paologalligit) [reply api/accounts/accounts.go:42]: How does `BlockCommitter` sound to you?

Comment (libotony) [reply api/accounts/accounts.go:42]: Do you think `bft.Committer` sounds good?

**Comment by paologalligit** [comment]
> can we add test for justified in e2e ? (we can open PR for now and just merge it when needed, before we forget about it)

Tests added in this [PR](https://github.com/vechain/thor-e2e-tests/pull/34).

**Comment by libotony** [comment]
@paologalligit I had non-persist way, if you have a bit, please take a look #808

**Review by darrenvechain — APPROVED** [review]

**Review by libotony — APPROVED** [review]

File: cmd/thor/node/packer_loop.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func (n *Node) pack(flow *packer.Flow) (err error) {
 
 		// write logs
 		if logEnabled {
-			if n.writeLogs(newBlock, receipts, oldBest.Header.ID()); err != nil {
+			if err := n.writeLogs(newBlock, receipts, oldBest.Header.ID()); err != nil {
```
Comment (otherview) [inline cmd/thor/node/packer_loop.go:161]: eh good spot

File: test/datagen/addess.go
Comment (otherview) [inline test/datagen/addess.go:1]: Missing an R in Address

**Review by otherview — COMMENTED** [review]
Did a partial review looks good, still want to digg one more time into the engine.go

File: api/utils/revisions.go [hunk 3]
```diff
@@ -41,6 +42,10 @@ func ParseRevision(revision string, allowNext bool) (*Revision, error) {
 		return &Revision{revFinalized}, nil
 	}
 
+	if revision == "justified" {
+		return &Revision{revJustified}, nil
+	}
+
```
[the remark this answers is not in the document]
Comment (paologalligit) [reply api/utils/revisions.go:48]: It's applicable everywhere the finalized revision is applicable. Basically it's a block you have a certain degree of confidence it won't be reverted, but not 100% sure like a finalized one

**Review by otherview — DISMISSED** [review]

**Review by darrenvechain — DISMISSED** [review]

**Review by otherview — APPROVED** [review]

**Review by libotony — APPROVED** [review]

**Pull request merged**